### PR TITLE
Migrate subscriptions engine to graphql-yoga

### PR DESCRIPTION
**Pull request by MichalLytek**
- use `@graphql-yoga` instead of `graphql-subscriptions` as the subscriptions engine
- require providing `PubSub` implementation into `buildSchema` option when using `@Subscription`
- remove `@PubSub` in favor of directly importing created `PubSub` implementation
- remove `Publisher` and `PubSubEngine` types
- add support for dynamic topic id function in `@Subscription` decorator option

### CODE REVIEW

**Comment by carlocorradini** [comment]
@MichalLytek Awesome.

What is missing?

**Comment by MichalLytek** [comment]
@carlocorradini Fixing pipeline issues, upgrading redis subscriptions example.

**Comment by carlocorradini** [comment]
Can I help somehow? ☺️

**Comment by carlocorradini** [comment]
@MichalLytek See PR #1590 🤗🥳

File: src/schema/schema-generator.ts [hunk 1]
```diff
@@ -635,48 +640,87 @@ export abstract class SchemaGenerator {
...
+  >(
     subscriptionsHandlers: SubscriptionResolverMetadata[],
-  ): GraphQLFieldConfigMap<T, U> {
+  ): GraphQLFieldConfigMap<TSource, TContext> {
+    if (!subscriptionsHandlers.length) {
+      return {};
+    }
     const { pubSub, container } = BuildContext;
+    if (!pubSub) {
+      // TODO: new error
```
Comment (MichalLytek) [inline src/schema/schema-generator.ts:654]: To be fixed

File: src/schema/schema-generator.ts [hunk 2]
```diff
@@ -635,48 +640,87 @@ export abstract class SchemaGenerator {
...
+            if (!Array.isArray(topics)) {
+              pubSubIterable = pubSub.subscribe(topics, topicId);
+            } else {
+              if (topics.length === 0) {
+                throw new MissingSubscriptionTopicsError(handler.target, handler.methodName);
+              }
+              // pubSubIterable = Repeater.merge([
+              //   undefined,
+              //   ...topics.map(topic => pubSub.subscribe(topic, topicId)),
+              // ]);
```
Comment (MichalLytek) [inline src/schema/schema-generator.ts:694]: to remove

**Pull request merged**